python/cudf/cudf/core/dataframe.py

```diff
@@ -5852,17 +5851,18 @@ def from_records(
     @_performance_tracking
     def _from_arrays(
         cls,
-        data: np.ndarray | cupy.ndarray,
+        data,
         index=None,
         columns=None,
         nan_as_null=False,
     ):
-        """Convert a numpy/cupy array to DataFrame.
+        """
+        Convert an object implementing an array interface to DataFrame.
 
         Parameters
         ----------
-        data : numpy/cupy array of ndim 1 or 2,
-            dimensions greater than 2 are not supported yet.
+        data : object of ndim 1 or 2,
+            Object implementing __array_interface__ or __cuda_array_interface__
```

Comment:
Might need this for formatting:
```suggestion
            Object implementing ``__array_interface__`` or ``__cuda_array_interface__``
```